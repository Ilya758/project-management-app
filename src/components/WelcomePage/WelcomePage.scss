@import url('https://fonts.googleapis.com/css2?family=Gideon+Roman&family=Playfair+Display:ital,wght@1,600&family=Prata&family=Roboto&family=Sansita+Swashed:wght@500&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Abril+Fatface&family=Gideon+Roman&family=Playfair+Display:ital,wght@1,600&family=Prata&family=Roboto&family=Sansita+Swashed:wght@500&display=swap');

.home-wrapper {
  display: flex;
  flex-direction: column;
  overflow: auto;

  & .about-wrapper {
    position: relative;
    height: 500px;
    background: url('../../assets/jpg/project-management-back.png') no-repeat center center;
    background-size: cover;
    background-attachment: fixed;
    display: flex;
    align-items: center;
    margin: 10px 0;

    & .title-primary-page {
      position: absolute;
      width: auto;
      height: auto;
      top: 5%;
      left: 5%;
      background: #030325fa;
      color: #61dafb;
      opacity: 0.7;
      padding: 10px;
      font-family: 'Open Sans', sans-serif;
      font-weight: 600;
      box-shadow: 6px 6px 3px 0 rgba(4, 25, 33, 0.73);
      font-size: 30px;
      letter-spacing: 5px;
      text-transform: uppercase;
    }

    & .second {
      top: 25%;
      left: 5%;
      font-size: 20px;
      text-transform: none;
      letter-spacing: 1px;
    }
  }

  & .project-info-content {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;

    & .project-info-content-card {
      width: 450px;
      margin: 15px;
      text-align: justify;
      height: auto;
      border-radius: 30px;
      display: flex;
      flex-direction: column;
      align-items: center;
      background-color: #f7f9fa;
      padding: 10px;
      box-shadow: 4px 4px 8px 0 rgba(35, 63, 84, 0.2);

      & div:first-child {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
      }

      & .imageFour {
        max-width: 100%;
        margin-bottom: 30px;
      }
    }
  }

  & .developers-content {
    align-items: center;
    display: flex;
    flex-direction: column;
    height: 1150px;

    & .developers-content-container {
      height: 100%;
      display: flex;
      flex-direction: column;
      justify-content: space-around;
      width: 100%;
      align-items: center;

      & .box-wrapper {
        width: 100%;
        max-width: 850px;
        border-radius: 40px;
        overflow: hidden;
        background-color: #edf1f2;
        height: 350px;
        padding: 10px 15px;
        display: flex;
        position: relative;
        justify-content: space-between;
        box-shadow: 4px 4px 8px 0 rgba(35, 63, 84, 0.2);
        margin: 20px 0;

        & .info-developer {
          padding: 10px;
          display: flex;
          flex-direction: column;
          align-items: center;
          justify-content: space-around;

          & .title-developer {
            font-size: 30px;
            text-align: center;
            color: #61dafb;
          }

          & .scills-developer {
            width: 100%;
            height: 60%;
            font-size: 20px;
            text-align: justify;
            color: #262637;
            font-family: 'ROBOTO';
            font-weight: 400;

            & p {
              margin: 15px 0;
            }
          }
        }

        & .image-developer {
          border-radius: 25px;
          width: 300px;
        }

        & .developer1 {
          background: url('../../assets/jpg/ilia.jpg') no-repeat center center;
          background-size: cover;
        }

        & .developer2 {
          background: url('../../assets/jpg/andrey.jpg') no-repeat center center;
          background-size: cover;
        }

        & .developer3 {
          background: url('../../assets/jpg/tatiana.jpg') no-repeat center center;
          background-size: cover;
        }
      }
    }
  }
}

.project-content {
  overflow: hidden;

  & .subheading {
    font-size: 32px;
    font-weight: 500;
    color: rgba(61, 71, 77, 0.5);
    letter-spacing: normal;
    text-align: center;
  }

  & div:first-child {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  & .primary-title {
    font-size: 80px;
    font-weight: 700;
    line-height: 100%;
    letter-spacing: -2px;
    text-align: center;

    & span:first-child {
      color: rgba(61, 71, 77, 0.5);
      line-height: 110%;
    }
  }

  & div {
    & .container-images {
      width: 100%;
      height: 500px;
      display: flex;
      justify-content: center;
      align-items: center;
      position: relative;

      & .imageOne {
        height: 100%;
      }

      & .imageTwo,
      .imageThree {
        position: absolute;
        max-width: 100%;
      }
    }
  }
}
