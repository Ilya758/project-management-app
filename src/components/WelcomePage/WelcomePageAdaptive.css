@media only screen and (min-width: 320px) and (max-width: 1160px) {
  .developers-content-container {
    flex-wrap: wrap;
    flex-direction: initial;
  }
  .developers-content {
    height: auto;
  }
  .box-wrapper {
    max-width: 550px;
    flex-direction: column;
    align-items: center;
    height: 600px;
    margin: 15px 0;
  }
  .image-developer {
    width: 400px;
    height: 450px;
  }
  .subheading {
    margin-bottom: 50px;
  }
}
