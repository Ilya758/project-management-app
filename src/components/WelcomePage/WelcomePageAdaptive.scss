@media only screen and (min-width: 320px) and (max-width: 1160px) {
  .home-wrapper {
    & .developers-content {
      height: auto;

      & .subheading {
        margin-bottom: 50px;
      }

      & .developers-content-container {
        flex-wrap: wrap;
        flex-direction: initial;

        & .box-wrapper {
          max-width: 550px;
          flex-direction: column;
          align-items: center;
          height: 600px;
          margin: 15px 0;

          & .image-developer {
            width: 400px;
            height: 450px;
          }
        }
      }
    }
  }
}

@media only screen and (min-width: 320px) and (max-width: 520px) {
  .home-wrapper {
    & .about-wrapper {
      background-attachment: inherit;

      & .title-primary-page {
        width: 65%;
      }
    }
  }
}

@media only screen and (min-width: 320px) and (max-width: 450px) {
  .logo {
    width: 42px;
    height: 42px;
    align-self: flex-start;
  }
}
