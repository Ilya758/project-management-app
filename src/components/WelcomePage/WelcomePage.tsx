import './WelcomePage.scss';
import ImageOne from '../../assets/jpg/heroWelcome.jpg';
import ImageTwo from '../../assets/svg/el1.svg';
import ImageThree from '../../assets/svg/el2.svg';
import ImageFour from '../../assets/svg/el4.svg';
import ImageFive from '../../assets/svg/el5.svg';
import ImageSix from '../../assets/svg/el6.svg';

export const WelcomePage = () => {
  return (
    <div className="home-wrapper">
      <section className="about-wrapper">
        <div className="title-primary-page">Project Management</div>
        <div className="title-primary-page second">...your unlimited use cases</div>
      </section>
      <section className="project-content">
        <div>
          <h1 className="subheading">Reliable Task Management for the Team</h1>
          <span className="primary-title">
            <span>Unite</span> Your Team
          </span>
          <div className="container-images">
            <img className="imageOne" src={ImageOne} alt="image" />
            <img className="imageTwo" src={ImageTwo} alt="image" />
            <img className="imageThree" src={ImageThree} alt="image" />
          </div>
        </div>
      </section>
      <section className="project-info-content">
        <div className="project-info-content-card">
          <img className="imageFour" src={ImageFour} alt="image" />
          <h3 className="card-info-title">TIME TO GET ORGANIZED</h3>
          <p>
            &nbsp; Task management is the link between the planning of cases and their
            implementation. The task management software needed to provide an overview of the work
            in order to translate the idea into a result.
            <br />
            &nbsp; Try <b>ProjectManagement:</b> Join the world as our teams use our teams to
            digitize workflows and get a clear overview of task progress. Let,s get organized
            together!
          </p>
        </div>
        <div className="project-info-content-card">
          <img className="imageFour" src={ImageFive} alt="image" />
          <h3 className="card-info-title">PROJECTS, THAT WORK</h3>
          <p>
            &nbsp; Whether you are managing your next big project or digitizing task management for
            your team is day-to-day work, you need to know who is doing what, when.{' '}
            <b>ProjectManagement</b> helps you manage tasks in a beautifully customizable
            environment that adapts perfectly to your needs.
          </p>
        </div>
        <div className="project-info-content-card">
          <img className="imageFour" src={ImageSix} alt="image" />
          <h3 className="card-info-title">DISTANT WORK</h3>
          <p>
            &nbsp; When your team is working from home, an efficient way to share information is
            essential to being productive. Use <b>ProjectManagement</b> to improve collaboration,
            manage tasks efficiently, and ensure projects run smoothly.
          </p>
        </div>
      </section>
      <section className="developers-content">
        <h2 className="subheading">Meet Team</h2>
        <div className="developers-content-container">
          <div className="box-wrapper">
            <div className="info-developer">
              <h3 className="title-developer">
                <em>Illia Skaryna</em> - Team lead, developer
              </h3>
              <div className="skills-developer">
                <p className="skills-developer-text">
                  - Basic project settings, application routing;
                </p>
                <p className="skills-developer-text">
                  - Support for the correct operation of the application;
                </p>
                <p className="skills-developer-text" />
                <p className="skills-developer-text" />
              </div>
            </div>
            <div className="image-developer developer1"></div>
          </div>
          <div className="box-wrapper">
            <div className="info-developer">
              <h3 className="title-developer">
                <em>Andrei Korozeev </em> - developer
              </h3>
              <div className="skills-developer">
                <p className="skills-developer-text">
                  - Registration and authorization of the user;
                </p>
                <p className="skills-developer-text" />
                <p className="skills-developer-text" />
                <p className="skills-developer-text" />
              </div>
            </div>
            <div className="image-developer developer2"></div>
          </div>
          <div className="box-wrapper">
            <div className="info-developer">
              <h3 className="title-developer">
                <em>Tatsiana Tuzava</em> - developer
              </h3>
              <div className="skills-developer">
                <p className="skills-developer-text">- Edit and delete profile of users;</p>
                <p className="skills-developer-text">- Sign Out from profile of users;</p>
                <p className="skills-developer-text" />
              </div>
            </div>
            <div className="image-developer developer3"></div>
          </div>
        </div>
      </section>
    </div>
  );
};
