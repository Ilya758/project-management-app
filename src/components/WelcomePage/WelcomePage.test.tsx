import { render } from '@testing-library/react';
import { WelcomePage } from '../WelcomePage/WelcomePage';

describe('WelcomePage Component', () => {
  it('Should render', () => {
    const component = render(<WelcomePage />);
    expect(component.baseElement).toMatchSnapshot();
  });
});
