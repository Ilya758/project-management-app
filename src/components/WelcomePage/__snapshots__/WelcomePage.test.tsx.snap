// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage Component Should render 1`] = `
<body>
  <div>
    <div
      class="home-wrapper"
    >
      <section
        class="about-wrapper"
      >
        <div>
          Project Management
        </div>
        <div>
          ... Ваши неограниченные варианты использования
        </div>
      </section>
      <section
        class="project-content"
      >
        <div>
          <h1
            class="subheading"
          >
            Надежное Управление Задачами для Команд
          </h1>
          <span
            class="primary-title"
          >
            <span>
              Объедините
            </span>
             Вашу Команду
          </span>
          <div
            class="container-images"
          >
            <img
              alt="image"
              class="imageOne"
              src="heroWelcome.jpg"
            />
            <img
              alt="image"
              class="imageTwo"
              src="el1.svg"
            />
            <img
              alt="image"
              class="imageThree"
              src="el2.svg"
            />
          </div>
        </div>
      </section>
      <section
        class="project-info-content"
      >
        <div
          class="card-info"
        >
          <img
            alt="image"
            class="imageFour"
            src="el4.svg"
          />
          <h3>
            Время Стать Организованными
          </h3>
          <p>
              Управление задачами является связующим звеном между планированием дел и их выполнением. Программное обеспечение для управления задачами должно обеспечивать обзор работы, чтобы видеть превращение идеи в результат.
            <br />
              Попробуйте 
            <b>
              ProjectManagement:
            </b>
             присоединяйтесь к командам со всего мира, которые используют наши доски для цифровизации рабочих процессов и получения четкого обзора прогресса задач. Станем организованными вместе!
          </p>
        </div>
        <div
          class="card-info"
        >
          <img
            alt="image"
            class="imageFour"
            src="el5.svg"
          />
          <h3>
            Проекты, Которые Работают
          </h3>
          <p>
              Независимо от того, управляете ли вы своим следующим крупным проектом или переводите управление задачами в цифровую форму для повседневной работы своей команды, вам нужно знать, кто, чем и когда занимается. 
            <b>
              ProjectManagement
            </b>
             поможет вам управлять задачами в красивой настраиваемой среде, которая идеально адаптируется к вашим потребностям.
          </p>
        </div>
        <div
          class="card-info"
        >
          <img
            alt="image"
            class="imageFour"
            src="el6.svg"
          />
          <h3>
            Удаленная Работа
          </h3>
          <p>
              Когда ваша команда работает из дома, эффективный способ обмена информацией имеет важное значение для продуктивной работы. Используйте 
            <b>
              ProjectManagement
            </b>
            , чтобы улучшить взаимодействие, эффективно управлять задачами и обеспечить гладкое выполнение проектов.
          </p>
        </div>
      </section>
      <section
        class="developers-content"
      >
        <h2
          class="subheading"
        >
          Наша команда
        </h2>
        <div
          class="developers-content-container"
        >
          <div
            class="card-info"
          >
            <h3>
              <em>
                Илья Скарына
              </em>
               - Team lead, developer
            </h3>
            <div
              class="image-developer developer1"
            />
            <p>
              - Базовые настройки проекта, роутинг приложения
            </p>
            <p>
              - Поддержка корректной работы приложения;
            </p>
            <p />
            <p />
          </div>
          <div
            class="card-info"
          >
            <h3>
              <em>
                Андрей Корозеев
              </em>
               - developer
            </h3>
            <div
              class="image-developer developer2"
            />
            <p>
              - Регистрация и авторизация пользователя;
            </p>
            <p />
            <p />
            <p />
          </div>
          <div
            class="card-info"
          >
            <h3>
              <em>
                Татьяна Тузова
              </em>
               - developer
            </h3>
            <div
              class="image-developer developer3"
            />
            <p>
              - Главная страница приложения;
            </p>
            <p />
            <p />
            <p />
          </div>
        </div>
      </section>
    </div>
  </div>
</body>
`;
